src/test/java/org/folio/rest/api/BoundWithStorageTest.java

```diff
@@ -0,0 +1,139 @@
+package org.folio.rest.api;
+
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import junitparams.JUnitParamsRunner;
+import org.folio.rest.support.IndividualResource;
+import org.folio.rest.support.Response;
+import org.folio.rest.support.builders.HoldingRequestBuilder;
+import org.folio.rest.support.builders.ItemRequestBuilder;
+import org.junit.After;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.net.HttpURLConnection;
+import java.util.List;
+import java.util.UUID;
+
+import static org.folio.rest.support.http.InterfaceUrls.*;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+@RunWith(JUnitParamsRunner.class)
+public class BoundWithStorageTest extends TestBaseWithInventoryUtil {
+
+  // see also @BeforeClass TestBaseWithInventoryUtil.beforeAny()
+
+  @After
+  public void afterEach() {
+    deleteBoundWithParts();
+    StorageTestSuite.deleteAll(itemsStorageUrl(""));
+    StorageTestSuite.deleteAll(holdingsStorageUrl(""));
+    StorageTestSuite.deleteAll(instancesStorageUrl(""));
+  }
+
+  @Test
+  public void canCreateAndRetrieveBoundWithParts() {
+    IndividualResource mainInstance = createInstance("Main Instance");
+    IndividualResource mainHoldingsRecord = createHoldingsRecord(mainInstance.getId());
+    IndividualResource item = createItem(mainHoldingsRecord.getId());
+
+    IndividualResource anotherInstance = createInstance("Another Instance");
+    IndividualResource anotherHoldingsRecord = createHoldingsRecord(anotherInstance.getId());
+    IndividualResource aThirdInstance = createInstance("Third Instance");
+    IndividualResource aThirdHoldingsRecord = createHoldingsRecord(aThirdInstance.getId());
+
+    // Make 'item' a bound-with
+    boundWithPartsClient.create(createBoundWithPartJson(mainHoldingsRecord.getId(),item.getId()));
```

Comment:
Does this mean that a bound-with item is intended to have a bound-with parts record for the primary holding as well as the secondary holdings?

Reply:
Yes.

Reply:
Though, it really comes down to how you decide to populate the data structure. You could decide to _not_  include the primary holding, but to fetch the complete set of titles in the bound-with, you would then have to remember to go out and find that directly linked holding separately.

Reply:
> Though, it really comes down to how you decide to populate the data structure. You could decide to _not_ include the primary holding, but to fetch the complete set of titles in the bound-with, you would then have to remember to go out and find that directly linked holding separately.

Who is `you` in these statements?

Are you suggesting that it will be up to each client to decide which way they want to do it, meaning that all clients would need to accommodate both situations when interpreting the data?

Reply:
Well, the business logic API can take care of always including all the holdings for an item and never include the primary holdings record more than once, regardless of whether the primary holdings record has an entry in the bound-with table. If we wanted that behavior just in case.

Clients that used the storage directly would have to pull the data out with an understanding of how it populated the data in the first place. To make the same client work for different population policies, against the raw storage, that client would need to do something similar to what the BL could do as described.

Reply:
> Well, the business logic API can take care of always including all the holdings for an item and never include the primary holdings record more than once, regardless of whether the primary holdings record has an entry in the bound-with table. If we wanted that behavior just in case.

That assumes that the clients will only be using the business logic APIs.

> Clients that used the storage directly would have to pull the data out with an understanding of how it populated the data in the first place. 

How would those clients (which might not be the clients that stored it) know how it was populated?

> To make the same client work for different population policies, against the raw storage, that client would need to do something similar to what the BL could do as described.

Yes, that is what I was suggesting, with this ambivalent approach, every client needs (the additional complexity) to be able to interpret either approach to structuring this data.